tests/core/pyspec/eth2spec/test/context.py

```diff
@@ -321,12 +321,23 @@ def wrapper(*args, **kw):
     return decorator
 
 
-def with_configs(configs):
+def disable_process_reveal_deadlines(fn):
+    def entry(*args, spec: Spec, **kw):
+        if hasattr(spec, 'process_reveal_deadlines'):
+            spec.process_reveal_deadlines = lambda state: None
```

Comment:
This mutates the spec object. Making it not re-usable. I.e. other tests afterwards will also have a no-operation `process_reveal_deadlines`. The decorator has to set the old function back, after running the wrapped test, for this to work as expected.

Also beware of returned generators not being evaluated directly; the wrapped function should be evaluated fully before undoing the no-op substitute. A similar thing is done in the BLS decorator, if you need an example. See `bls_switch` in `context.py`

Reply:
Thanks for the catch! 👍👍
I forgot that the test generator mode scope != pytest unit test mode scope.